src/main/java/com/sensimity/android/Sensimity.java

```diff
@@ -0,0 +1,92 @@
+package com.sensimity.android;
+
+import android.content.Context;
+
+public class Sensimity {
+    protected static Sensimity instance = null;
+
+    protected static Context context = null;
+    protected static String username = null;
+    protected static String password = null;
+    protected static String clientId = null;
+    protected static String clientSecret = null;
+    protected static String instanceRef = null;
+
+    /**
+     * Initialize the Sensimity library with credentials.
+     *
+     * @param context      Application context
+     * @param username     Sensimity username
+     * @param password     Sensimity password
+     * @param clientId     Sensimity oAuth client id
+     * @param clientSecret Sensimity oAuth client secret
+     * @param instanceRef  Instance ref used for sending the scan results
+     * @return Sensimity instance
+     */
+    public static void initialize(Context context, String username, String password, String clientId, String clientSecret, String instanceRef) {
+        if (instance == null) {
+            instance = new Sensimity();
```

Comment:
What is the reason to make Sensimity a singleton, while it could be regular object referencing to itself?